app/com/linkedin/drelephant/tunin/BaselineComputeUtil.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.tunin;
+
+import java.util.List;
+
+import models.TuningJobDefinition;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.log4j.Logger;
+
+import play.libs.Json;
+
+import com.avaje.ebean.Ebean;
+import com.avaje.ebean.SqlRow;
+import com.linkedin.drelephant.ElephantContext;
+import com.linkedin.drelephant.util.Utils;
+
+
+/**
+ * This class does baseline computation once the job is enabled for auto tuning.
+ * It takes average resource usage and execution time for last 30 jobs to get the baseline.
+ */
+public class BaselineComputeUtil {
+
+  public static Integer numJobsForBaselineDefault = 30;
+  public Integer numJobsForBaseline = null;
+  private final Logger logger = Logger.getLogger(getClass());
+  public static final String BASELINE_EXECUTION_COUNT = "baseline.execution.count";
+
+  public BaselineComputeUtil() {
+    Configuration configuration = ElephantContext.instance().getAutoTuningConf();
+    numJobsForBaseline = Utils.getNonNegativeInt(configuration, BASELINE_EXECUTION_COUNT, numJobsForBaselineDefault);
+  }
+
+  public List<TuningJobDefinition> computeBaseline() {
+    try {
+      List<TuningJobDefinition> tuningJobDefinitions = getJobForBaselineComputation();
+      for (TuningJobDefinition tuningJobDefinition : tuningJobDefinitions) {
+        updateBaselineForJob(tuningJobDefinition);
+      }
+      return tuningJobDefinitions;
+    } catch (Exception e) {
+      logger.error("Error in computing baseline for ", e);
+    }
+    return null;
+  }
+
+  public List<TuningJobDefinition> getJobForBaselineComputation() {
+    logger.info("Starting Computing baseline for jobs: ");
+
+    List<TuningJobDefinition> tuningJobDefinitions =
+        TuningJobDefinition.find.where().eq(TuningJobDefinition.TABLE.averageResourceUsage, null).findList();
+    logger.debug("Computing baseline for jobs: " + Json.toJson(tuningJobDefinitions));
+    logger.info("Baseline computing finished.");
+    return tuningJobDefinitions;
+  }
+
+  public void updateBaselineForJob(TuningJobDefinition tuningJobDefinition) {
```

Comment:
Why can't these functions be part of TuningJobDefinition instead of a separate util class ?

Reply:
In general POJO's should contain definition of the object, not much logic, so it's better to keep this as a different utility.